staging/src/k8s.io/legacy-cloud-providers/vsphere/vsphere.go

```diff
@@ -1070,31 +1066,33 @@ func (vs *VSphere) DisksAreAttached(nodeVolumes map[k8stypes.NodeName][]string)
 				VC_DC := nodeInfo.vcServer + nodeInfo.dataCenter.String()
 				dcNodes[VC_DC] = append(dcNodes[VC_DC], nodeName)
 			}
-
+		
+			rNodes := make(chan []k8stypes.NodeName)
+			ech := make(chan error)
```

Comment:
Good point. Thank you.

Reply:
@misterikkit thanks for the valuable insight. Can we instead create a buffered channel of slice(s) to avoid blocking on the channel ?

"This could also leak goroutines if one returns an error and the other is blocked trying to write to rNodes" . Could you elaborate more on this ? I think I am missing the return statement right after 'ech <- err' statement so that one goroutine writes to either the error channel or rNodes channel but not to both.